batik-svg-dom/src/main/java/org/apache/batik/dom/svg/AbstractSVGTransformList.java

```diff
@@ -284,22 +284,21 @@ public void assign(SVGTransform transform) {
                     setScale(matrix.getA(), matrix.getD());
                     break;
                 case SVGTransform.SVG_TRANSFORM_ROTATE:
-                    if (matrix.getE() == 0.0f) {
-                        rotate(transform.getAngle());
-                    } else {
-                        angleOnly = false;
-                        if (matrix.getA() == 1.0f) {
-                            setRotate(transform.getAngle(),
-                                      matrix.getE(), matrix.getF());
-                        } else if (transform instanceof AbstractSVGTransform) {
-                            AbstractSVGTransform internal =
-                                (AbstractSVGTransform) transform;
+                    angleOnly = false;
+                    if (transform instanceof AbstractSVGTransform) {
+                        AbstractSVGTransform internal =
+                            (AbstractSVGTransform) transform;
+                        if (internal.getX() == 0.0f && internal.getY() == 0.0f) {
+                            rotate(internal.getAngle());
+                        } else {
                             setRotate(internal.getAngle(),
                                       internal.getX(), internal.getY());
```

Comment:
I guess I could simplify this further like:

```suggestion
                        angleOnly = internal.getX() == 0.0f
                                    && internal.getY() == 0.0f;
                        setRotate(internal.getAngle(),
                                  internal.getX(), internal.getY());
```